### PR TITLE
Adding passwordless SSH and fixing some of the spec files

### PR DESCRIPTION
**Pull request by miroswan**
* Adding passwordless SSH
* Fixing issue with legacy include statements in default spec tests.

### CODE REVIEW

**Comment by coderanger** [comment]
Love the idea, but we can't do this with a static private key :-/ The security issues would be too dangerous. Any chance you could change this to generate a new key pair using something like https://stackoverflow.com/questions/5270386/generate-ssh-keypairs-private-public-without-ssh-keygen ?

**Comment by miroswan** [comment]
I was actually thinking about generating the key on the host and injecting the public key into the container. I think ssh-keygen would have to be installed on the host, which might be problematic for Windows users. Perhaps there is another way. I'll take a look at that stackoverflow link.

**Comment by coderanger** [comment]
Yeah, you can do it all in Ruby code using the `ssh_type` and `to_blob` methods mentioned in that post.

**Comment by coderanger** [comment]
Also if you want to be really fancy, you could use something like https://github.com/coderanger/kitchen-sync/blob/master/lib/kitchen-sync/rsync.rb#L46 to grab an existing identity from ssh-agent and then fall back to creating a new one if that fails.

**Comment by miroswan** [comment]
I have some new code to start. I'm using the sshkey library to generate the pairs. Let me know what you think. I'm assuming it could use some graceful error handling. At least it's a working draft.

**Comment by coderanger** [comment]
Given this can be done easily with net-ssh already, I'm not sure it is worth the extra dependency.

**Comment by miroswan** [comment]
Kinda like how easy sshkey makes it, but I can take a look at net-ssh soonish.

**Comment by miroswan** [comment]
Using net/ssh now.

**Comment by jeffbyrnes** [comment]
:+1: to this, I’ve tested and it fixes #110 and #81 as far as I can tell.

**Comment by coderanger** [comment]
Love it. @portertech any thoughts on overall thumbs up/down? If not I'll merge this soon.

File: lib/kitchen/driver/docker.rb [hunk 1]
```diff
@@ -126,6 +131,22 @@ def docker_command(cmd, options={})
         run_command("#{docker} #{cmd}", options.merge(:quiet => !logger.debug?))
       end
 
+      def generate_keys
+        if ! File.exist?(config[:public_key]) or ! File.exist?(config[:private_key])
```
Comment (portertech) [inline lib/kitchen/driver/docker.rb]: Ruby style? Maybe this should be `!File.exist?()`?

Comment (portertech) [inline lib/kitchen/driver/docker.rb]: Or use `unless` :+1:

File: lib/kitchen/driver/docker.rb [hunk 2]
```diff
@@ -181,6 +204,11 @@ def build_dockerfile
           RUN mkdir -p /etc/sudoers.d
           RUN echo '#{username} ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers.d/#{username}
           RUN chmod 0440 /etc/sudoers.d/#{username}
+          RUN [ ! -d /home/kitchen/.ssh ] && mkdir /home/kitchen/.ssh
+          RUN chown -R kitchen:kitchen /home/kitchen/.ssh
+          RUN chmod 0700 /home/kitchen/.ssh
+          RUN echo '#{IO.read(config[:public_key])}' >> /home/kitchen/.ssh/authorized_keys
```
Comment (portertech) [inline lib/kitchen/driver/docker.rb]: Maybe we should use a variable, something along the lines of `public_key` or `public_key_contents`? Assign it prior to `base =`.

File: lib/kitchen/driver/docker.rb (hunk 1 above)
Comment (coderanger) [inline lib/kitchen/driver/docker.rb]: Yeah, also needs `||` instead of `or` but I can fix that up post merge.

File: lib/kitchen/driver/docker.rb [hunk 3]
```diff
@@ -181,6 +204,11 @@ def build_dockerfile
...
+          RUN chown kitchen:kitchen /home/kitchen/.ssh/authorized_keys && chmod 0600 /home/kitchen/.ssh/authorized_keys
```
Comment (portertech) [inline lib/kitchen/driver/docker.rb]: Maybe use two separate `RUN` commands? Easier to read and no run-on.

**Comment by miroswan** [comment]
@coderanger Did you want me to make these adjustments? I don't mind.

**Comment by miroswan** [comment]
I updated. I chose not to go with unless because it's sometimes unclear if the negation applies to both. Did correct the style though.

**Comment by portertech** [comment]
@miroswan Thank you :+1: Will try to get this into a release before the weekend!

**Pull request merged**